### PR TITLE
Add typings for Typescript

### PR DESCRIPTION
**Pull request by pachuka**
Based this off of the documentation at: https://docs.heapanalytics.com/reference

### CODE REVIEW

**Comment by drob** [comment]
This is awesome, thank you for the PR! 🎉🎉🎉

We'll take a look promptly.

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
+/**
+ * Sets the app ID where you'll be sending data. It can be used to switch between projects or
+ * between your production and development environments.
+ *
+ * `setAppId` initializes Heap tracking on iOS. It should be called only once per app open.
+ * If it is called multiple times, it will initiate a new session for the user.
```
Comment (drob) [inline index.d.ts]: If it's called more than once in a react native iOS app, subsequent calls will be ignored. @GrandPoohBear is that the behavior on android or will it create a new session?

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+ *
+ * @param appId the appId corresponding to one of your projects
+ */
+export function setAppId(appId: string): void;
+
+/**
+ * Send additional events to Heap.
+ *
+ * For Android, the properties map may be null to track an event with no properties.
```
Comment (drob) [inline index.d.ts]: Is this specific to android?

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+export function addUserProperties(properties: object): void;
+
+/**
+ * Specify a set of global key-value pairs to get attached to all of a user's subsequent events.
+ *
+ * These event properties will persist across multiple sessions on the same device and get applied to both
+ * auto-captured and custom events.
+ *
+ * This is useful if you have some persistent state, but you don't want to apply it across all of a user's
+ * events with `identify`. A good example is "Logged In", which changes over the user's lifecycle.
```
Comment (drob) [inline index.d.ts]: Should be `addUserProperties` here, not `identify`.

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+ */
+export function removeEventProperty(property: string): void;
+
+/**
+ * Removes all stored event properties.
+ */
+export function clearEventProperties(): void;
+
+/**
+ * Changes the frequency with which events are sent back to the Heap server from an iOS mobile device.
```
Comment (drob) [inline index.d.ts]: I think this is a copy-paste mistake. I don't think we expose `changeInterval` yet.

File: index.d.ts (hunk 1 above)
Comment (GrandPoohBear) [reply index.d.ts]: Yeah, unfortunately this behavior is different across platforms right now, but it's something we will have to address from the native lib side.  Current behavior as of 0.1.0:
iOS - Subsequent calls will be ignored, so only the first appId will ever be used (by default, the native lib stops sending events if this gets called twice, so the bridge prevents it.)
Android - This initializes an entirely new tracker object, so the session will be created from scratch.

File: index.d.ts (hunk 4 above)
Comment (GrandPoohBear) [reply index.d.ts]: The old React Native library had exposed it on iOS, but Android only had it as a private function (IIRC).  To keep the same interface, we dropped that from the initial version.

Comment (pachuka) [reply index.d.ts]: My mistake, just removed it.

File: index.d.ts (hunk 1 above)
Comment (pachuka) [reply index.d.ts]: I updated it with a note about the differences between behavior on ios vs android.

File: index.d.ts (hunk 2 above)
Comment (pachuka) [reply index.d.ts]: Removed the 'For Android, '

**Pull request merged**